File: LeetCode/[M] - 714. Best Time to Buy and Sell Stock with Transaction Fee/solution.go
package main

func maxProfit(prices []int, fee int) int {
	hold, cash := 0-prices[0], 0

	for i := 1; i < len(prices); i++ {
		hold, cash = max(hold, cash-prices[i]), max(cash, hold+prices[i]-fee)
	}
	return cash
}

File: LeetCode/[M] - 714. Best Time to Buy and Sell Stock with Transaction Fee/solution.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int], fee: int) -> int:
        hold, cash = 0 - prices[0], 0

        for i in range(1, len(prices)):
            hold, cash = (
                max(hold, cash - prices[i]),
                max(cash, hold + prices[i] - fee),
            )

        return cash
